### PR TITLE
[HttpKernel] Improve and add tests for `Last-Modified` computation with ESI responses

### LINKED ISSUE #41666: HttpCache should provide `Last-Modified` for ESI responses when possible
Currently, `ResponseCacheStrategy` will always remove the `Last-Modified` header when merging ESI responses. This makes it impossible to revalidate the result.

My suggestion is to keep the maximum = most recent `Last-Modified` value on the final response as long as all (sub-)responses provided `Last-Modified`.

Does that make sense? Why not? Am I missing edge cases?

Ping @dbu and @Toflar since I know you're familiar with HTTP Caching, and @aschempp since you wrote a good amount of the code involved.

### PR DESCRIPTION
**Pull request by mpdude**
| Q             | A
| ------------- | ---
| Branch?       | 6.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | 
| License       | MIT

This PR slightly improves the test cases for #42355 (which fixed #41666).

It adds two tests for the cases where `Last-Modified` should _not_ be added,
and updates the existing test to make it more obvious that we only look at
the existence of `Last-Modified` headers, whereas the `ETag` is not
relevant.

### CODE REVIEW

**Comment by mpdude** [comment]
Inviting @aschempp as the author of #42355 for a review.

Background: I'm currently investigating a bug where ESI responses miss content. It might be that the change from #42355 – which I think is correct in itself :+1: – creates the condition where the bug appears.

File: src/Symfony/Component/HttpKernel/Tests/HttpCache/ResponseCacheStrategyTest.php [hunk 1]
```diff
@@ -138,22 +138,48 @@ public function testLastModifiedIsMergedWithEmbeddedResponse()
...
+    }
+
+    public function testLastModifiedIsRemovedWhenEmbeddedResponseHasNoLastModified()
+    {
+        $cacheStrategy = new ResponseCacheStrategy();
+
+        $mainResponse = new Response();
+        $mainResponse->setLastModified(new \DateTimeImmutable('-2 hour'));
 
-        // Embedded response uses "expiry" model
```
Comment (mpdude) [inline src/Symfony/Component/HttpKernel/Tests/HttpCache/ResponseCacheStrategyTest.php:148]: I think "expiry" refers to the `maxAge`, but we're only using `Last-Modified` in both reqests. `Last-Modified` is, to my understanding, one possibly way of revalidation.

**Review by aschempp — APPROVED** [review]
Tests look good to me. I can't honestly tell why I added the `Etag` to the main response, since (as you said) this should not be relevant at all.

> Background: I'm currently investigating a bug where ESI responses miss content. It might be that the change from #42355 – which I think is correct in itself 👍 – creates the condition where the bug appears.

Sounds like `HttpCache` returns a `304` response without content (which is according to specs), while trying to merge ESI fragments into the main response 🤔

**Comment by mpdude** [comment]
> Sounds like `HttpCache` returns a `304` response without content (which is according to specs), while trying to merge ESI fragments into the main response 🤔

See #58015.

**Comment by nicolas-grekas** [comment]
Thank you @mpdude.

**Pull request merged**